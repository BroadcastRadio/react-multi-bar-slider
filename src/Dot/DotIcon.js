import PropTypes from 'prop-types';
import glamorous from 'glamorous';

const DotIcon = glamorous.img({
  position: 'absolute',
  transform: 'translateX(-50%)',
  userDrag: 'none',
  userSelect: 'none'
}, ({ width, height }) => ({
  width,
  height
}));

DotIcon.displayName = 'DotIcon';

DotIcon.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  css: PropTypes.object
};

DotIcon.defaultProps = {
  width: 50,
  height: 50
};

export default DotIcon;
