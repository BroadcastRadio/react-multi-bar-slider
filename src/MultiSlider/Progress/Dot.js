import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import transition from '../utils/transition';

const StyledDot = glamorous.span({
  position: 'absolute',
  display: 'block',
  zIndex: 5,
  borderRadius: '50%'
}, ({ hasIcon, width, height, color, noTransition }) => ({
  top: hasIcon ? 0 : '50%',
  transform: hasIcon ? 'translateX(-50%)' : 'translate(50%, -50%)',
  width: hasIcon ? 0 : width,
  height: hasIcon ? 0 : height,
  backgroundColor: hasIcon ? 'transparent' : color,
  transition: noTransition ? 'none' : transition
}));

const DotIcon = glamorous.img({
  position: 'absolute',
  transform: 'translateX(-50%)',
  userDrag: 'none',
  userSelect: 'none'
}, ({ width, height }) => ({
  width,
  height
}));

const Dot = ({ dot, sliderColor, mouseDown }) => (
  <StyledDot
    className="dot"
    hasIcon={!!dot.icon}
    width={dot.width || 28}
    height={dot.height || 28}
    color={dot.color || sliderColor}
    css={dot.style}
    noTransition={mouseDown}
  >
    {dot.icon && (
      <DotIcon
        className="icon"
        src={dot.icon}
        width={dot.width || 50}
        height={dot.height || 50}
        draggable={false}
        css={dot.iconStyle}
      />
    )}
  </StyledDot>
);

Dot.propTypes = {
  dot: PropTypes.oneOfType([
    PropTypes.shape({
      width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      color: PropTypes.string,
      icon: PropTypes.string,
      style: PropTypes.object
    }),
    PropTypes.bool
  ]).isRequired,
  sliderColor: PropTypes.string.isRequired,
  mouseDown: PropTypes.bool.isRequired
};

export default Dot;
