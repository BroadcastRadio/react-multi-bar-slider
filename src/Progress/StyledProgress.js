import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import transition from '../utils/transition';
import getHalf from '../utils/getHalf';

const StyledProgress = glamorous.div(
  {
    position: 'absolute',
    top: 0,
  },
  ({
    color,
    progress,
    height,
    equal,
    equalColor,
    roundedCorners,
    reversed,
    noTransition,
    zIndex
  }) => ({
    left: reversed ? 'auto' : 0,
    right: reversed ? 0 : 'auto',
    width: `${progress || 0}%`,
    height,
    backgroundColor: equal && equalColor ? equalColor : color,
    borderRadius: roundedCorners ? getHalf(height) : 0,
    zIndex,
    transition: noTransition ? 'none' : transition
  })
);

StyledProgress.propTypes = {
  color: PropTypes.string.isRequired,
  progress: PropTypes.number.isRequired,
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  equal: PropTypes.bool.isRequired,
  equalColor: PropTypes.string,
  roundedCorners: PropTypes.bool.isRequired,
  reversed: PropTypes.bool.isRequired,
  noTransition: PropTypes.bool.isRequired,
  zIndex: PropTypes.number.isRequired,
  css: PropTypes.object
};

StyledProgress.defaultProps = {
  equal: false
};

export default StyledProgress;
