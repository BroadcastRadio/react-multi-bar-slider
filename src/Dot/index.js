import React from 'react';
import PropTypes from 'prop-types';
import processStyle from '../utils/processStyle';
import StyledDot from './StyledDot';
import DotIcon from './DotIcon';

const Dot = ({
  width,
  height,
  color,
  icon,
  style,
  iconStyle,
  progressColor,
  reversed,
  mouseDown,
  children,
  ...props
}) => (
  <StyledDot
    className="dot"
    hasIcon={!!icon}
    width={width}
    height={height}
    color={color || progressColor}
    reversed={reversed}
    noTransition={mouseDown}
    css={processStyle(style, {
      width,
      height,
      color,
      icon,
      progressColor,
      reversed,
      mouseDown
    })}
    {...props}
  >
    {icon && (
      <DotIcon
        className="icon"
        src={icon}
        width={width}
        height={height}
        draggable={false}
        css={processStyle(iconStyle, {
          width,
          height,
          color,
          icon,
          progressColor,
          mouseDown
        })}
      />
    )}
    {children}
  </StyledDot>
);

Dot.displayName = 'Dot';

Dot.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  color: PropTypes.string,
  icon: PropTypes.string,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
  iconStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
  progressColor: PropTypes.string,
  reversed: PropTypes.bool,
  mouseDown: PropTypes.bool,
  children: PropTypes.node
};

export default Dot;
