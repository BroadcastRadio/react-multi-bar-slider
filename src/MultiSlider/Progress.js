import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import sliderPropType from './utils/sliderPropType';
import glamorous from 'glamorous';
import transition from './utils/transition';
import getHalf from './utils/getHalf';
import Dot from './Dot';

const StyledProgress = glamorous.div(
  {
    position: 'absolute',
    top: 0,
    right: 0
  },
  ({
    color,
    progress,
    height,
    equal,
    equalColor,
    roundedCorners,
    zIndex,
    noTransition
  }) => ({
    width: `${progress || 0}%`,
    height,
    backgroundColor: equal && equalColor ? equalColor : color,
    borderRadius: roundedCorners ? getHalf(height) : 0,
    zIndex,
    transition: noTransition ? 'none' : transition
  })
);

const Progress = ({
  slider,
  height,
  slidersEqual,
  equalColor,
  roundedCorners,
  mouseDown,
  zIndex
}) => (
  <Fragment>
    <StyledProgress
      className="progress"
      color={slider.color}
      progress={slider.progress}
      height={height}
      equal={slidersEqual}
      equalColor={equalColor}
      roundedCorners={roundedCorners}
      zIndex={zIndex}
      noTransition={mouseDown}
    />
    {slider.dot && (
      <Dot
        dot={slider.dot}
        progress={slider.progress}
        sliderColor={slider.color}
        mouseDown={mouseDown}
      />
    )}
  </Fragment>
);

Progress.propTypes = {
  slider: sliderPropType,
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  slidersEqual: PropTypes.bool.isRequired,
  equalColor: PropTypes.string.isRequired,
  roundedCorners: PropTypes.bool.isRequired,
  mouseDown: PropTypes.bool.isRequired,
  zIndex: PropTypes.number.isRequired
};

export default Progress;
