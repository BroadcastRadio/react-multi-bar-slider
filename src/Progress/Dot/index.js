import React from 'react';
import PropTypes from 'prop-types';
import processStyle from '../../utils/processStyle';
import StyledDot from './StyledDot';
import DotIcon from './DotIcon';

const Dot = ({ dot, sliderColor, reversed, mouseDown }) => (
  <StyledDot
    className="dot"
    hasIcon={!!dot.icon}
    width={dot.width}
    height={dot.height}
    color={dot.color || sliderColor}
    reversed={reversed}
    noTransition={mouseDown}
    css={processStyle(dot.style, { dot, sliderColor, reversed, mouseDown })}
  >
    {dot.icon && (
      <DotIcon
        className="icon"
        src={dot.icon}
        width={dot.width }
        height={dot.height}
        draggable={false}
        css={processStyle(dot.iconStyle, { dot, sliderColor, mouseDown })}
      />
    )}
  </StyledDot>
);

Dot.propTypes = {
  dot: PropTypes.oneOfType([
    PropTypes.shape({
      width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      color: PropTypes.string,
      icon: PropTypes.string,
      style: PropTypes.object
    }),
    PropTypes.bool
  ]).isRequired,
  sliderColor: PropTypes.string.isRequired,
  reversed: PropTypes.bool.isRequired,
  mouseDown: PropTypes.bool.isRequired
};

export default Dot;
