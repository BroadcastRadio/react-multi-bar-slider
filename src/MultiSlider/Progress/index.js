import React from 'react';
import PropTypes from 'prop-types';
import sliderPropType from '../utils/sliderPropType';
import Dot from './Dot';
import processStyle from '../utils/processStyle';
import StyledProgress from './StyledProgress';

const Progress = ({
  slider,
  height,
  slidersEqual,
  equalColor,
  roundedCorners,
  mouseDown,
  zIndex
}) => (
  <StyledProgress
    className="progress"
    color={slider.color}
    progress={slider.progress}
    height={height}
    equal={slidersEqual}
    equalColor={equalColor}
    roundedCorners={roundedCorners}
    noTransition={mouseDown}
    zIndex={zIndex}
    css={processStyle(slider.style, {
      slider,
      height,
      slidersEqual,
      equalColor,
      roundedCorners,
      mouseDown,
      zIndex
    })}
  >
    {slider.dot && (
      <Dot
        dot={slider.dot}
        sliderColor={slider.color}
        mouseDown={mouseDown}
      />
    )}
  </StyledProgress>
);

Progress.propTypes = {
  slider: sliderPropType,
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  slidersEqual: PropTypes.bool.isRequired,
  equalColor: PropTypes.string,
  roundedCorners: PropTypes.bool.isRequired,
  mouseDown: PropTypes.bool.isRequired,
  zIndex: PropTypes.number.isRequired
};

export default Progress;
