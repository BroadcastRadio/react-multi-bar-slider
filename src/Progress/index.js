import React, { Children } from 'react';
import PropTypes from 'prop-types';
import processStyle from '../utils/processStyle';
import StyledProgress from './StyledProgress';

const Progress = ({
  color,
  progress,
  style,
  height,
  progressEqual,
  equalColor,
  roundedCorners,
  reversed,
  mouseDown,
  zIndex,
  children,
  ...props
}) => (
  <StyledProgress
    className="progress"
    color={color}
    progress={progress}
    height={height}
    equal={progressEqual}
    equalColor={equalColor}
    roundedCorners={roundedCorners}
    reversed={reversed}
    noTransition={mouseDown}
    zIndex={zIndex}
    css={processStyle(style, {
      color,
      progress,
      height,
      progressEqual,
      equalColor,
      roundedCorners,
      reversed,
      mouseDown,
      zIndex
    })}
    {...props}
  >
    {children && Children.map(children, child =>
      React.cloneElement(child, {
        progressColor: color,
        reversed,
        mouseDown
      })
    )}
  </StyledProgress>
);

Progress.displayName = 'Progress';

Progress.propTypes = {
  color: PropTypes.string.isRequired,
  progress: PropTypes.number.isRequired,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  progressEqual: PropTypes.bool,
  equalColor: PropTypes.string,
  roundedCorners: PropTypes.bool,
  reversed: PropTypes.bool,
  mouseDown: PropTypes.bool,
  zIndex: PropTypes.number,
  children: PropTypes.node
};

export default Progress;
