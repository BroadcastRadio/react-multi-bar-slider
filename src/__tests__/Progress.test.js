import React from 'react';
import Enzyme, { shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Progress from '../Progress';
import StyledProgress from '../Progress/StyledProgress';
import Dot from '../Progress/Dot';

Enzyme.configure({ adapter: new Adapter() });

beforeEach(() => {
  global.progressProps = {
    slider: {
      color: 'blue',
      progress: 39
    },
    height: 14,
    slidersEqual: true,
    equalColor: 'purple',
    roundedCorners: false,
    reversed: false,
    mouseDown: false,
    zIndex: 4
  };
});

afterEach(() => {
  delete global.progressProps;
});

/* global progressProps */
describe('Progress.js', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(<Progress {...progressProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const progress = render(<Progress {...progressProps} />);
    expect(progress).toBeDefined();
  });

  it('renders a StyledProgress', () => {
    const progress = shallow(<Progress {...progressProps} />);
    expect(progress.find(StyledProgress).length).toBe(1);
  });

  it('renders a dot when given', () => {
    const progress = shallow(<Progress {...progressProps} />);
    expect(progress.find(Dot).length).toBe(0);

    progressProps.slider.dot = { color: 'grey' };
    const progress2 = shallow(<Progress {...progressProps} />);
    expect(progress2.find(Dot).length).toBe(1);
    expect(progress2.find(Dot).prop('dot')).toEqual({ color: 'grey' });
  });
});
