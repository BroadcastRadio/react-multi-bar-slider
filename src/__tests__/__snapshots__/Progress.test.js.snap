// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress.js matches the snapshot 1`] = `
<div
  className="css-fc62n7 progress"
/>
`;
